R/bench-summ.R

```diff
@@ -0,0 +1,166 @@
+#' @importFrom rlang .data
+get_bench_summary <- function(.res_tibble) {
+  # get roc results
+  roc <- format_roc(.res_tibble, "roc")
+
+  # get PR roc results
+  pr <- format_roc(.res_tibble, "prc")
+
+  # Plot ROC
+  roc_plot <- ggplot(roc, aes(x = 1-specificity,
+                              y = sensitivity,
+                              colour = run_key)) +
+    geom_line() +
+    geom_abline(intercept = 0, slope = 1, linetype = "dashed") +
+    xlab("FPR (1-specificity)") +
+    ylab("TPR (sensitivity)")
+
+  # Plot PR ROC
+  pr_plot <- ggplot(pr, aes(x = recall, y = precision , colour = run_key)) +
+    geom_line() +
+    geom_abline(intercept = 0, slope = 1, linetype = "dashed") +
+    xlab("Recall/Sensitivity") +
+    ylab("Precision")
+
+  # Extract AUROC
+  auroc_tibble <- .res_tibble %>%
+    unnest(roc) %>%
+    select(set_name, bench_name, filter_crit, statistic, auc) %>%
+    distinct()
+
+  # Plot AUROC
+  auroc_plot <- auroc_tibble %>%
+    unite("run_key", set_name, bench_name,
+          statistic, filter_crit, remove = F) %>%
+    ggplot(., aes(x = reorder(run_key, auc),
+                  y = auc,
+                  fill = run_key)) +
+    geom_bar(stat = "identity") +
+    xlab("networks") +
+    ylab("AUROC") +
+    coord_flip(ylim = c(0.5, 0.8)) +
+    theme(legend.position = "none")
+
+  # AUROC Heatmap
+  auroc_heat <- auroc_tibble %>% get_auroc_heat()
+
+  # Extract AU PRROC
+  prauc_tibble <- .res_tibble %>%
+    unnest(prc) %>%
+    select(set_name, bench_name, filter_crit, statistic, auc) %>%
+    distinct()
+
+  # AU PR Heatmap
+  pr_heat <- prauc_tibble %>% get_auroc_heat()
+
+  # get computational time info
+  comp_time <- .res_tibble %>%
+    # get statistic time from activity
+    mutate(statistic_time = activity %>%
+             map(function(tib)
+               tib %>%
+                 select(statistic_time) %>%
+                 unique)) %>%
+    unnest(statistic_time) %>%
+    # calculate regulon size
+    group_by(set_name, bench_name, filter_crit) %>%
+    mutate(regulon_time = sum(statistic_time)) %>%
+    select(set_name, bench_name, statistic,
+           filter_crit, statistic_time, regulon_time)
+
+  # Join AUROC, PRAUC, Coverage, and Comp time
+  summary_table <- auroc_tibble %>%
+    inner_join(prauc_tibble %>%
+                 rename(pr_auc = auc),
+               by = c("set_name", "bench_name", "statistic", "filter_crit")) %>%
+    distinct() %>%
+    inner_join(x=.,
+               y=(roc %>%
+                    group_by(name_lvl) %>%
+                    summarise(source_cov = coverage,
+                              condition_cov = n,
+                              roc_neg=tn) %>%
+                    distinct() %>%
+                    ungroup() %>%
+                    separate(col="name_lvl",
+                             into=c("set_name", "bench_name", "filter_crit"),
+                             sep="\\.")),
+               by = c("set_name", "bench_name", "filter_crit")) %>%
```

Comment:
using pipes is a good thing but this goes a bit too far in my opinion. Create the objects you would like to join separately